modules/codegen/src/main/scala/com/twilio/guardrail/generators/Java/JacksonGenerator.scala

```diff
@@ -0,0 +1,580 @@
+                ))
+              }): _*
+            )
+          )
+        )
+
+        // TODO: handle emptyToNull in the return for the getters
+        terms.foreach({ case ParameterTerm(_, parameterName, fieldType, _) =>
+          val method = dtoClass.addMethod(s"get${parameterName.capitalize}", PUBLIC)
+          method.setType(fieldType)
+          method.setBody(new BlockStmt(new NodeList(
+            new ReturnStmt(new FieldAccessExpr(new ThisExpr, parameterName))
+          )))
+        })
+
+        val builderMethod = dtoClass.addMethod("builder", PUBLIC)
+        builderMethod.setType("Builder")
+        builderMethod.setParameters(new NodeList(
+          requiredTerms.map({ case ParameterTerm(_, parameterName, _, parameterType) =>
+            new Parameter(util.EnumSet.of(FINAL), parameterType, new SimpleName(parameterName))
+          }): _*
+        ))
+        builderMethod.setBody(new BlockStmt(new NodeList(
+          new ReturnStmt(
+            new ObjectCreationExpr(
+              null,
+              JavaParser.parseClassOrInterfaceType("Builder"),
+              new NodeList(requiredTerms.map({
+                case ParameterTerm(_, parameterName, _, _) => new NameExpr(parameterName)
+              }): _*)
+            )
+          )
+        )))
+
+        val builderClass = new ClassOrInterfaceDeclaration(util.EnumSet.of(PUBLIC, STATIC), false, "Builder")
+
+        requiredTerms.foreach({ case ParameterTerm(_, parameterName, fieldType, _) =>
+          builderClass.addField(fieldType, parameterName, PRIVATE, FINAL)
+        })
+        optionalTerms.foreach({ case ParameterTerm(_, parameterName, fieldType, _) =>
+          // TODO: initialize with default if one is specified
+          builderClass.addFieldWithInitializer(fieldType, parameterName, new MethodCallExpr("java.util.Optional.empty"), PRIVATE)
+        })
+
+        val builderConstructor = builderClass.addConstructor(PRIVATE)
+        builderConstructor.setParameters(new NodeList(
+          requiredTerms.map({ case ParameterTerm(_, parameterName, _, parameterType) =>
+            new Parameter(util.EnumSet.of(FINAL), parameterType, new SimpleName(parameterName))
+          }): _*
+        ))
+        builderConstructor.setBody(new BlockStmt(new NodeList(
+          requiredTerms.map({ case ParameterTerm(_, parameterName, fieldType, _) =>
+            new ExpressionStmt(
+              new AssignExpr(
+                new FieldAccessExpr(new ThisExpr, parameterName),
+                fieldType match {
+                  case _: PrimitiveType => new FieldAccessExpr(new NameExpr("builder"), parameterName)
+                  case _ => new MethodCallExpr("requireNonNull", new FieldAccessExpr(new NameExpr("builder"), parameterName))
+                },
+                AssignExpr.Operator.ASSIGN
+              )
+            )
+          }): _*
+        )))
+
+        // TODO: leave out with${name}() if readOnlyKey?
```

Comment:
"read only" means different things from the server side and client side. I haven't come up with a good way to represent this yet. We should want either use case to work correctly, and getting the response from a server and giving it to a client should work without resetting the read-only values to null.